### DISCUSSION TITLE
Borg Backup pre&post backup script / webhook for mainenance notifications

### DISCUSSION THREAD

**Discussion by tau-ri** [nextcloud/all-in-one#4217] [Ideas] [answered]
Requested Feature: Webhook / API / SCRIPT execution BEFORE and AFTER Borg Backup.
Why: To avoid alerting of any Monitoring solution during planned maintenance.

Explanation:
I'm using Nextcloud AIO  and BORG backup what is running fine!
For monitoring, I'm using [Uptime Kuma ](https://github.com/louislam/uptime-kuma) and because there is no build in API (yet), I have [Uptime-Kuma-Web-API ](https://github.com/MedAziz11/Uptime-Kuma-Web-API) attached. This is running fine and giving me the possibility to do a lot of automation! Kudos to [@louisalm ](https://github.com/louislam) and [@MedAziz11 ](https://github.com/MedAziz11)

While doing (daily) BORG backups, I need to set MAINTENANCE to the Uptime Kuma Nextcloud monitors.
As per now, I've setup a schedule in Uptime Kuma what is in alignment with START time of Borg backup. But the time varies for the backups and hence, the End Date is always different. I do not want to extend the END time to anything.

So I'm looking into a solution within NC AIO to set a PRE and POST backup execution.

**Comment by szaimen** [comment, marked as answer]
Hi, as this sounds like a nice feature, the requests for this are quite low. Currently there are no plans to implement such a feature. 

So I don't see a better way for you currently than extending the End time in uptime Kuma.

**Reply by tau-ri** [reply]
Thanks @szaimen for your answer and time.
Anyways I'm going then to implement something for my scenario and once ready I'll love to share it with the community.

**Comment by tau-ri** [comment]
I see in the nextcloud-aio-mastercontainer LOGS the line "Daily backup script has finished" what I assume indicates that the backup fully completed?
```
Waiting for backup container to stop
Starting and updating containers...
Waiting for the Nextcloud container to start
Waiting for the Nextcloud container to start
Sending backup notification...
Daily backup script has finished
Total reclaimed space: 0B
```

If so, I'll use this as a trigger.

**Reply by szaimen** [reply]
> "Daily backup script has finished" what I assume indicates that the backup fully completed?

yes
